types/src/account_address.rs

```diff
@@ -46,16 +48,18 @@ impl AccountAddress {
         self.0.to_vec()
     }
 
-    pub fn authentication_key<PublicKey: VerifyingKey>(
-        public_key: &PublicKey,
-    ) -> AuthenticationKey {
+    pub fn authentication_key(public_key: &Ed25519PublicKey) -> AuthenticationKey {
         AuthenticationKey::from_public_key(public_key)
     }
 
-    pub fn from_public_key<PublicKey: VerifyingKey>(public_key: &PublicKey) -> Self {
+    pub fn from_public_key(public_key: &Ed25519PublicKey) -> Self {
```

Comment:
So the decision is to not allow generics?

Reply:
That's part of the crypto issue, I guess I was wondering if we wanted to at least use type aliasing, but whatever `sed` is a powerful tool, we don't need to be too clever.

https://github.com/libra/libra/issues/2683

I'm actually enjoying my afternoon doing a little bit of work in this space, so this is good enough...